markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -121,9 +122,17 @@ library PerpsMarket {
         return (orderFeeData.makerFee + orderFeeData.takerFee).mulDecimal(marketConfig.skewScale);
     }
 
-    function updatePositionData(Data storage self, Position.Data memory newPosition) internal {
-        self.size += MathUtil.abs(newPosition.size);
-        self.skew += newPosition.size;
+    function updatePositionData(
+        Data storage self,
+        uint128 accountId,
+        Position.Data memory newPosition
+    ) internal {
+        Position.Data storage oldPosition = self.positions[accountId];
+        int128 oldPositionSize = oldPosition.size;
+
+        self.size = MathUtil.abs(oldPositionSize + newPosition.size);
```

Comment:
I think this is wrong.. The `size` lives on the market, so I assume that's the total size of the market? Here we setting the whole market size to `oldPositionSize + newPosition.size`. 
Something like this would make sense to me:
`self.size += MathUtil.abs(newPosition.size) - MathUtil.abs(newPosition.size);` 

I would also be in favour of changing this name to open interest

Reply:
oh wow yeah totally got this wrong lol; while i'm at it while write a test 👍🏽